### ISSUE TITLE
TS support

### ISSUE THREAD

**Issue by rafaelss95** [not-an-aardvark/eslint-plugin-eslint-plugin#176]
Hello ✋

I was trying to use this library in a project, but as I didn't receive any reports (even though I knew there were violations there), I decided to look at the source code and I saw that some rules like `no-missing-placeholders` and `report-message-format` (those are the rules I was most interested in, but I believe there are much more) basically depend on `utils.getContextIdentifiers`, which looks for `module.exports` or `exports.create` (**please, correct if I'm wrong  — I have no familiarity with this repository code and this assumption comes from the little I could see by looking at the code + tests**), but with TS (more specifically `@typescript-eslint`), the rules are _usually_ created using the following structure:

```ts
// https://github.com/timdeschryver/eslint-plugin-ngrx
export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
  meta: {
    // ...
  }
})

// https://github.com/angular-eslint/angular-eslint
export default createESLintRule<Options, MessageIds>({
  meta: {
    // ...
  }
})

// https://github.com/typescript-eslint/typescript-eslint/issues
export default util.createRule<Options, MessageIds>({
  meta: {
    // ...
  }
})
```

So I wonder if the implementation could be changed to not depend specifically on `module.exports` or `exports.create` and to support these cases above as well. I really wanted to avoid redundancy and use some of the rules that already exist here instead of rewriting from scratch, but I understand if it's not within the scope of your project.

**Comment by bmish** [comment]
I would be in favor of supporting TypeScript rules. I guess we could just check that the first argument to the `export default` function call (e.g. `util.createRule`) has an object with `create`/`meta`.

Relatedly, we also want to support ESM rules (https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/issues/157) which are similar except without the TypeScript helper.

**Comment by bmish** [comment]
I've opened a PR to support this. Will be doing more testing of it soon -- feel free to test it as well.
